### PR TITLE
LPS-161162 Migrate folder, article and template actions to dropdown action items

### PR DESCRIPTION
**Pull request by adolfopa**

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:stop

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/info_panel.jsp [hunk 1]
```diff
@@ -67,7 +67,15 @@ if (ListUtil.isEmpty(kbFolders) && ListUtil.isEmpty(kbArticles)) {
 				<div class="autofit-col">
 					<ul class="autofit-padded-no-gutters autofit-row">
 						<li class="autofit-col">
-							<liferay-util:include page="/admin/kb_folder_action.jsp" servletContext="<%= application %>" />
+
```
Comment (AliciaGarciaGarcia) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/info_panel.jsp:70]: There is some kind of trouble with the actions from the info_panel (maybe the redirection) that previously did not happened. Can you take a look?

https://user-images.githubusercontent.com/47524751/186184995-225cca48-2199-45c1-bb80-4075e85c6733.mp4

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/util/KBDropdownItemsProvider.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+		_liferayPortletRequest = liferayPortletRequest;
+		_liferayPortletResponse = liferayPortletResponse;
+
+		_themeDisplay = (ThemeDisplay)liferayPortletRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+	}
+
+	public List<DropdownItem> getKBArticleDropdownItems(KBArticle kbArticle) {
+		return DropdownItemListBuilder.add(
+			dropdownItem -> {
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/util/KBDropdownItemsProvider.java:66]: This is coming from 
```
<c:if test="<%= KBArticlePermission.contains(permissionChecker, kbArticle, KBActionKeys.UPDATE) %>">
		<liferay-portlet:renderURL var="editURL">
			<portlet:param name="mvcPath" value="/admin/common/edit_kb_article.jsp" />
			<portlet:param name="redirect" value="<%= currentURL %>" />
			<portlet:param name="resourcePrimKey" value="<%= String.valueOf(kbArticle.getResourcePrimKey()) %>" />
		</liferay-portlet:renderURL>

		<liferay-ui:icon
			message="edit"
			url="<%= editURL %>"
		/>
	</c:if>
```

But there is no check for the permissions (you are missing something like `this::_hasUpdatePermission` ??)

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBArticleDropdownPropsTransformer.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {openConfirmModal} from 'frontend-js-web';
+
+const ACTIONS = {
+	delete(itemData) {
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBArticleDropdownPropsTransformer.js]: better `delete({deleteKBArticleURL})`

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBFolderDropdownPropsTransformer.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			dialogIframe: {
+				bodyCssClass: 'dialog-with-footer',
+			},
+			title: Liferay.Language.get(label),
+			uri: url,
+		});
+	},
+
+	permissions(itemData) {
+		this.openWindow(
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBFolderDropdownPropsTransformer.js]: Same [here](https://github.com/liferay-lima/liferay-portal/pull/3402/commits/ed13afab83eb072223dc3dd5af23e71a055a2b1b#r953480901)

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBFolderDropdownPropsTransformer.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {openConfirmModal} from 'frontend-js-web';
+
+const ACTIONS = {
+	delete(itemData) {
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBFolderDropdownPropsTransformer.js]: Simplify destructuring itemData with `delete({deleteKBArticleURL})`, is the pattern we use to follow.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/info_panel.jsp (hunk 1 above)
Comment (adolfopa) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/info_panel.jsp:70]: Fixed.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBArticleDropdownPropsTransformer.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+			dialogIframe: {
+				bodyCssClass: 'dialog-with-footer',
+			},
+			title: Liferay.Language.get(label),
+			uri: url,
+		});
+	},
+
+	permissions(itemData) {
+		this.openWindow(
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBArticleDropdownPropsTransformer.js]: Done.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBArticleDropdownPropsTransformer.js (hunk 3 above)
Comment (adolfopa) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBArticleDropdownPropsTransformer.js]: Done.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBFolderDropdownPropsTransformer.js (hunk 4 above)
Comment (adolfopa) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBFolderDropdownPropsTransformer.js]: Done.

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
Added a fix to avoid extra padding in article's dropdown. also checked the issue that @AliciaGarciaGarcia mentioned and it is fixed, so I'm forwarding the pull. Thanks @adolfopa

**Comment by ambrinchaudhary** [comment]
ci:forward

**Pull request closed without merging**